### PR TITLE
Change premerge to build all Spark versions

### PR DESCRIPTION
**Pull request by tgravescs**
Build all the versions to catch bugs in not updating shims.

### CODE REVIEW

**Comment by tgravescs** [comment]
build

**Review by gerashegalov — DISMISSED** [review]

**Review by razajafri — APPROVED** [review]

**Review by jlowe — DISMISSED** [review]

**Comment by tgravescs** [comment]
premerge failed due to cdh compile issue, I'll wait til that is fixed and retest this

**Comment by tgravescs** [comment]
build

File: jenkins/spark-premerge-build.sh [hunk 1]
```diff
@@ -35,18 +35,24 @@ mvn_verify() {
...
+    # don't skip tests
+    env -u SPARK_HOME mvn -U -B $MVN_URM_MIRROR -Dbuildver=311 clean install -Drat.skip=true -Dmaven.javadoc.skip=true -Dskip -Dmaven.scalastyle.skip=true -Dcuda.version=$CUDA_CLASSIFIER -Dpytest.TEST_TAGS=''
+    env -u SPARK_HOME mvn -U -B $MVN_URM_MIRROR -Dbuildver=311cdh clean install -Drat.skip=true -DskipTests -Dmaven.javadoc.skip=true -Dskip -Dmaven.scalastyle.skip=true -Dcuda.version=$CUDA_CLASSIFIER -pl aggregator -am
+    env -u SPARK_HOME mvn -U -B $MVN_URM_MIRROR -Dbuildver=312 clean install -Drat.skip=true -DskipTests -Dmaven.javadoc.skip=true -Dskip -Dmaven.scalastyle.skip=true -Dcuda.version=$CUDA_CLASSIFIER -pl aggregator -am
+    env -u SPARK_HOME mvn -U -B $MVN_URM_MIRROR -Dbuildver=313 clean install -Drat.skip=true -DskipTests -Dmaven.javadoc.skip=true -Dskip -Dmaven.scalastyle.skip=true -Dcuda.version=$CUDA_CLASSIFIER -pl aggregator -am
+    # don't skip tests
+    env -u SPARK_HOME mvn -U -B $MVN_URM_MIRROR -Dbuildver=320 clean install -Drat.skip=true -Dmaven.javadoc.skip=true -Dskip -Dmaven.scalastyle.skip=true -Dcuda.version=$CUDA_CLASSIFIER -pl -Dpytest.TEST_TAGS=''
```
Comment (gerashegalov) [inline jenkins/spark-premerge-build.sh:49]: CI catches `aggregator` is missing here

**Comment by tgravescs** [comment]
build

File: jenkins/spark-premerge-build.sh (hunk 1 above)
Comment (tgravescs) [reply jenkins/spark-premerge-build.sh]: actually removed -pl for this one because running tests with it so want to build everything

**Review by gerashegalov — APPROVED** [review]

**Comment by tgravescs** [comment]
switch the profile back to individual from snapshots. somehow it appears to be downloading jars to the deps directory with timestamp for cdh jar instead of snapshot, so I'll followup with that separately.

**Comment by tgravescs** [comment]
build

**Review by gerashegalov — APPROVED** [review]

**Pull request merged**